### PR TITLE
Check existence of JSON data from both catalog and LMS course API

### PR DESCRIPTION
**Pull request by mjfrey**
Always check if a JSON attribute is avail before accessing it. We can not assume the data is always correct.

### CODE REVIEW

**Comment by mjfrey** [comment]
@mattdrayer

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py [hunk 1]
```diff
@@ -352,3 +352,37 @@ def test_get_course_offer_data(self):
             'title': course.name,
             'voucher_end_date': voucher.end_datetime
         })
+
+    def test_get_course_offer_null_image_data(self):
```
Comment (mattdrayer) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py]: Since we're also testing the null start date scenario here, it probably makes sense to rename this method to something more generic like `test_get_course_offer_verify_null_fields` or something along those lines.  Then you can just add tack an in-line comment onto the end of each assertion.

File: ecommerce/extensions/api/v2/tests/views/test_vouchers.py [hunk 2]
```diff
@@ -352,3 +352,37 @@ def test_get_course_offer_data(self):
...
+
+        self.assertDictEqual(offer, {
+            'benefit': {
+                'type': benefit.type,
+                'value': benefit.value
+            },
+            'contains_verified': True,
+            'course_start_date': '',
+            'id': course.id,
+            'image_url': '',
```
Comment (mattdrayer) [inline ecommerce/extensions/api/v2/tests/views/test_vouchers.py]: I'd append a comment to this line to highlight that this is one of the pertinent checks.  Another way to do this would be to have specific assertions for each field we're concerned with -- it's valid to do the full-on dictionary check but then it's not really clear what we're asserting.

File: ecommerce/extensions/basket/tests/test_views.py [hunk 3]
```diff
@@ -397,3 +397,18 @@ def test_cached_course(self):
         self.assertEqual(response.status_code, 200)
         cached_course_after = cache.get(cache_hash)
         self.assertEqual(cached_course_after['name'], self.course.name)
+
+    def test_empty_course_api_response(self):
+        seat = self.create_seat(self.course)
+        self.create_basket_and_add_product(seat)
+        course_info_json = json.dumps('{}')
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/tests/test_views.py]: Nit: Move below the course_url setup on line 406 so it's closer to where it's being used.

File: ecommerce/extensions/basket/tests/test_views.py [hunk 4]
```diff
@@ -397,3 +397,18 @@ def test_cached_course(self):
...
+        course_id = 'course-v1:test+test+test'
+        course_url = get_lms_url('api/courses/v1/courses/{}/'.format(course_id))
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/tests/test_views.py:406]: Is this the same as self.path below?

File: ecommerce/extensions/basket/tests/test_views.py [hunk 5]
```diff
@@ -397,3 +397,18 @@ def test_cached_course(self):
         self.assertEqual(response.status_code, 200)
         cached_course_after = cache.get(cache_hash)
         self.assertEqual(cached_course_after['name'], self.course.name)
+
+    def test_empty_course_api_response(self):
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/tests/test_views.py:401]: It's not clear to me what is being validated based on the name of this test -- if the name is correct maybe add a docstring describing the use case(s) being covered?

File: ecommerce/extensions/basket/tests/test_views.py [hunk 6]
```diff
@@ -397,3 +397,18 @@ def test_cached_course(self):
...
+        seat = self.create_seat(self.course)
+        self.create_basket_and_add_product(seat)
+        course_info_json = json.dumps('{}')
+        course_id = 'course-v1:test+test+test'
+        course_url = get_lms_url('api/courses/v1/courses/{}/'.format(course_id))
+        httpretty.register_uri(httpretty.GET, course_url, body=course_info_json, content_type='application/json')
+
+        response = self.client.get(self.path)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context['formset_lines_data']), 1)
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/tests/test_views.py]: Is it really important for this particular test to verify that only one row is being returned?  Does it matter?

File: ecommerce/extensions/basket/views.py [hunk 7]
```diff
@@ -107,8 +107,8 @@ def get_context_data(self, **kwargs):
             short_description = None
             try:
                 course = get_course_info_from_lms(course_key)
-                image_url = get_lms_url(course['media']['course_image']['uri'])
-                short_description = course['short_description']
+                image_url = course.get('media', '{}').get('image', '{}').get('raw', '')
+                short_description = course.get('short_description', '')
                 course_name = course['name']
             except (ConnectionError, SlumberBaseException, Timeout):
```
Comment (mattdrayer) [inline ecommerce/extensions/basket/views.py:113]: The try is only protecting against errors stemming from `get_course_info_from_lms`, so the except should be moved up to that line -- the rest can all come after.

**Comment by mattdrayer** [comment]
A couple clarification requests and some nits, otherwise LGTM -- 👍

File: ecommerce/extensions/basket/tests/test_views.py (hunk 4 above)
Comment (mjfrey) [reply ecommerce/extensions/basket/tests/test_views.py:406]: no
